macos-cxx17.cmake

```diff
@@ -0,0 +1,21 @@
+# Copyright (c) 2016-2018, Ruslan Baratov
+# Copyright (c) 2017, David Hirvonen
+# All rights reserved.
+
+if(DEFINED POLLY_CLANG_CXX17_CMAKE_)
+  return()
+else()
+  set(POLLY_CLANG_CXX17_CMAKE_ 1)
+endif()
+
+include("${CMAKE_CURRENT_LIST_DIR}/utilities/polly_init.cmake")
+
+polly_init(
+    "clang / c++17 support"
+    "Unix Makefiles"
+)
+set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}  -pthread " CACHE STRING "")
```

Comment:
I think you don't need to forcefully select pthread on mac and linux.

Reply:
After test you have right we don't need ```-pthread ``` on macOS and Linux